src/coll/mod.rs

```diff
@@ -319,7 +328,10 @@ where
         resolve_options!(self, options, [write_concern]);
 
         let delete = Delete::new(self.namespace(), query, Some(1), options);
-        self.client().execute_operation(delete).await
+        self.client()
+            .execute_operation(delete)
```

Comment:
It could reduce on boilerplate a little bit to have a separate `execute_cursor_operation` so that all these other methods which could never make use of the returned connection don't have to deal with it.

Reply:
There's already an `execute_cursor_operation` and it does something different

Reply:
Oh right, I forgot we already added something similar before. While it does currently do something different, it actually needs to be expanded and used for exhaust cursors too. Otherwise, the implicit session used by an exhaust cursor will get returned to the session pool as soon as the initial command is executed, and it could then be used by other operations while the cursor is still alive (i.e. concurrently).

Reply:
In the latest commit I pushed, I think this is handled properly. The session used for the initial command is passed into the cursor constructor and kept around for as long as the cursor is alive, with a reference being passed in to each getMore operation. I moved it into the `CursorResponse` type (which was called `OperationResult` in the first commit of this PR) rather than having a tuple of `(OperationResult, Option<Session>)` being returned and then having to pass both elements in separately to the constructor.